src/OpenTelemetry.AutoInstrumentation/Instrumentation.cs

```diff
@@ -137,6 +167,12 @@ public static void Initialize()
                     Logger.Information("Initialized lazily-loaded metric instrumentations without initializing sdk.");
                 }
             }
+#if NET6_0_OR_GREATER
+            if (profilerEnabled && (threadSamplingEnabled || allocationSamplingEnabled))
```

Comment:
Is there a reason this is done last? Could this be moved to the earlier conditional blocks?

Reply:
Technically yes, but for traces and metrics we would like to take all possible data (at least for AlwaysOn Sampler).
For the profiler, we are heavily dropping data/sampling once per second(s). So dropping data is acceptable, both at the beggining and end of the process.